sled-agent/src/sled_agent.rs

```diff
@@ -306,13 +307,16 @@ pub enum InventoryError {
     BadByteCount(#[from] ByteCountRangeError),
     #[error(transparent)]
     InventoryError(#[from] sled_agent_config_reconciler::InventoryError),
+    #[error(transparent)]
+    ExecutionError(#[from] illumos_utils::ExecutionError),
```

Comment:
Do we need to add an error variant here? I thought if we got back an error from the health checkers, it'd result in a _successful_ inventory containing that error as a `String`?

Reply:
Good catch! I did this a while ago and then forgot I added it